src/python_testing/TC_OCC_3_2.py

```diff
@@ -1,204 +1,240 @@
+        time.sleep(0.001)
+
+    @async_test_body
+    async def test_TC_OCC_3_2(self):
+        endpoint_id = self.matter_test_config.endpoint
+        node_id = self.dut_node_id
+        dev_ctrl = self.default_controller
+
+        post_prompt_settle_delay_seconds = 10.0
+        cluster = Clusters.Objects.OccupancySensing
+        attributes = cluster.Attributes
+
+        # CI app pipe id creation
+        self.app_pipe = "/tmp/chip_all_clusters_fifo_"
+        self.is_ci = self.check_pics("PICS_SDK_CI_ONLY")
+        if self.is_ci:
+            app_pid = self.matter_test_config.app_pid
+            if app_pid == 0:
+                asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set.c")
+            self.app_pipe = self.app_pipe + str(app_pid)
+
+        self.step(1)  # Commissioning already done
+
+        self.step(2)
+        feature_map = await self.read_occ_attribute_expect_success(attribute=attributes.FeatureMap)
+        has_feature_pir = (feature_map & cluster.Bitmaps.Feature.kPassiveInfrared) != 0
+        has_feature_ultrasonic = (feature_map & cluster.Bitmaps.Feature.kUltrasonic) != 0
+        has_feature_contact = (feature_map & cluster.Bitmaps.Feature.kPhysicalContact) != 0
+
+        logging.info(
+            f"Feature map: 0x{feature_map:x}. PIR: {has_feature_pir}, US:{has_feature_ultrasonic}, PHY:{has_feature_contact}")
+
+        attribute_list = await self.read_occ_attribute_expect_success(attribute=attributes.AttributeList)
+        has_pir_timing_attrib = attributes.PIROccupiedToUnoccupiedDelay.attribute_id in attribute_list
+        has_ultrasonic_timing_attrib = attributes.UltrasonicOccupiedToUnoccupiedDelay.attribute_id in attribute_list
+        has_contact_timing_attrib = attributes.PhysicalContactOccupiedToUnoccupiedDelay.attribute_id in attribute_list
+        logging.info(f"Attribute list: {attribute_list}")
+        logging.info(f"--> Has PIROccupiedToUnoccupiedDelay: {has_pir_timing_attrib}")
+        logging.info(f"--> Has UltrasonicOccupiedToUnoccupiedDelay: {has_ultrasonic_timing_attrib}")
+        logging.info(f"--> Has PhysicalContactOccupiedToUnoccupiedDelay: {has_contact_timing_attrib}")
+
+        # min interval = 0, and max interval = 30 seconds
+        attrib_listener = ClusterAttributeChangeAccumulator(Clusters.Objects.OccupancySensing)
+        await attrib_listener.start(dev_ctrl, node_id, endpoint=endpoint_id, min_interval_sec=0, max_interval_sec=30)
+
+        # add Namepiped to assimilate the manual sensor untrigger here
+        self.step("3a")
+        # CI call to trigger off
+        if self.is_ci:
+            self.write_to_app_pipe('{"Name":"SetOccupancy", "EndpointId": 1, "Occupancy": 0}')
+        else:
+            self.wait_for_user_input(prompt_msg="Type any letter and press ENTER after DUT goes back to unoccupied state.")
+
+        self.step("3b")
+        if attributes.Occupancy.attribute_id in attribute_list:

```

Comment:
This is always true, so let's remove this if.

Reply:
removed it.